pkg/common/utils/utils.go

```diff
@@ -58,6 +58,10 @@ func NeedRecovery(pod *v1.Pod) (bool, error) {
 		return true, nil
 	}
 
+	if pod.Status.Phase == v1.PodFailed && pod.Status.Reason == constants.ApplicationRejectedFailure || pod.Status.Reason == constants.ApplicationInsufficientResourcesFailure {
+		return false, nil
+	}
```

Comment:
Can we simply return false here when pod phase is Failed? 
Do we have to check the reason?

Reply:
You are right. Removed reason check